osbrain/tests/test_agent_async_requests.py

```diff
@@ -22,114 +23,110 @@ def on_error(agent):
 class Server(Agent):
     def on_init(self):
         self.received = []
+        self.blocked = False
 
 
 class Client(Agent):
     def on_init(self):
         self.received = []
 
 
-@pytest.fixture(scope='function')
-def server_client_late_reply_return():
-    def late_reply(agent, request):
-        agent.received.append(request)
-        time.sleep(1)
-        return 'x' + request
-
+def async_server_client(server_handler=append_received,
+                        client_handler=append_received):
```

Comment:
You are not using a different `client_handler` anywhere, so I'd remove it from this function for simplicity.